aima-core/src/main/java/aima/core/agent/DTAgent.java

```diff
@@ -0,0 +1,76 @@
+package aima.core.agent;
+
+
+import aima.core.agent.Action;
+import aima.core.agent.Percept;
+import aima.core.agent.impl.AbstractAgent;
+import aima.core.probability.BeliefState;
+
+/**
+ * Artificial Intelligence A Modern Approach (3rd Edition): Figure 13.1, page 484.<br>
+ * 
+ * Figure 13.1  A decision-theoretic agent that selects rational actions.
+ * <br>
+ *
+ * <pre>
+ * function DT-AGENT(percept) returns an action
+ *   persistent: belief_state, probabilistic beliefs about the current state of the world
+ *               action, the agent's action
+ *
+ *   update belief_state based on action and percept
+ *   calculate outcome probabilities for actions
+ *      given action descriptions and current belief state
+ *   select action with highest expected utility
+ *      given probabilities of outcomes and utility information
+ *   return action
+ *
+ * </pre>
+ *
+ *  @author Ritwik Sharma
+ *  @author samagra
+ */
+public abstract class DTAgent extends AbstractAgent {
+	/**
+	 * @param belief_state
+	 *            	      probabilistic beliefs about the current state of the world.
+	 * @param action
+	 *              the agent's action
+	 */
+	BeliefState<Action,Percept> belief_state;
+	Action action;
+	Action action_descriptions;
+
+	public Action perceive(Percept percept) {
+
+		// update belief_state based on action and percept
+		belief_state.update(action,percept);
+		/** 
+		 * calculate outcome probabilities for actions,
+	   	 *		given action descriptions and current belief_state 
+		*/	
+		Action action_probabilities = cal_action_probabilities(action_descriptions,belief_state);
+		/**  
+		 * select action with highest expected utility
+	   	 *		given probabilities of outcomes and utility information
+		 */
+		Action finalAction = action_with_highest_expected_utility(action_probabilities);
+

```

Comment:
@ritwik12 Here, you are giving this function a single Action. How can he select an action with the highest utility from a single action? There is nothing to choose from.